report/4.Design/RTS.tex

```diff
@@ -6,7 +6,74 @@ \subsection{Scheduling}\label{Design:Scheduling}
+Using the UPPAAL tool, a model of the tasks of the system was constructed. 
+
+\paragraph{Templates}
+When creating an UPPAAL project, a templates must be made, which is a model that describes a system or a part of a system.  
+The templates are later used to make up the entire system, and they can communicate using channels specified in the declarations of the project.
+
+\paragraph{Locations}
+An UPPAAL model consists of at least one \textit{location}, which is a state that the system can be in. 
+A location can optionally be specified as \textit{initial}, \textit{urgent}, \textit{committed}. 
+If a location is marked as initial, it is defined as the starting state for the modelled system. 
+A location marked with urgent will be the preferred destination, if there are multiple possible transitions. 
+A location marked as committed is an location that is preferred over an location without any type and that is overruled by a location marked as urgent. 
+If a location is not marked with an specific type, the location might be taken but has the lowest priority if any other transitions are possible. 
+A location can be given an \textit{invariant} that can be used to specify a condition that must be fulfilled before a location can be left. 
+
+\paragraph{Edges}
+Between locations an edge can be created, they are used to model a shift in model state and is used to control the flow through the template. 
+On edges different methods can be used to enforce a wanted behavior and transfer information between templates. 
+
+\paragraph{Declarations}
+Once the templates has been created, the declaration is used to make the entire system function. 
+The declaration is can contain variables, channels and methods used in the templates and between templates. 
+Declarations can be specified on a template level and on an global level. 
+
+\paragraph{System declarations}
+The system model in the templates is initialized from the system declarations, where instances of the templates are initialized and added to the system as can be seen on \autoref{snip:UPPAALSystemDeclaration}. 
+
+
+\begin{lstlisting}[label={snip:UPPAALSystemDeclaration},caption={System declaration from UPPAAL},frame=tlrb,numbers=none]
+BackgroundTask    = Task( 0 );
+UpdateDisplayTask = Task( 1 );
+MoveMotorsTask    = Task( 2 );
+ReciveDataTask    = Task( 3 );
+HandleLaserTask   = Task( 4 );
+KeepUSBAliveTask  = Task( 5 );
+
+system BackgroundTask, UpdateDisplayTask, MoveMotorsTask, ReciveDataTask, HandleLaserTask, KeepUSBAliveTask, Scheduler;
+\end{lstlisting}
+Each task in initialized and assigned to an new task which is then added to the system along with the scheduler. 
+
+On \autoref{fig:UppaalTask} the behavior of the a task in can be seen. 
+The model is a timed automaton which is used to ensure that the task will come true to its deadline and periods. 
```

Comment:
```suggestion
The model is a timed automaton, which is used to ensure that the task will fulfil its deadline and periods. 
```

Reply:
tror faktisk det skal være be true to its deadline